src/panels/calendar/show-confirm-event-dialog-box.ts

```diff
@@ -0,0 +1,35 @@
+import { TemplateResult } from "lit";
+import { fireEvent } from "../../common/dom/fire_event";
+import { RecurrenceRange } from "../../data/calendar";
+
+export interface ConfirmEventDialogBoxParams {
+  confirmText?: string;
+  confirmFutureText?: string; // Prompt for future recurring events
+  confirm?: (recurrenceRange: RecurrenceRange) => void;
+  cancel?: () => void;
+  text?: string | TemplateResult;
+  title: string;
+}
+
+export const loadGenericDialog = () => import("./confirm-event-dialog-box");
+
+export const showConfirmEventDialog = (
+  element: HTMLElement,
+  dialogParams: ConfirmEventDialogBoxParams
+) =>
+  new Promise<RecurrenceRange | undefined>((resolve) => {
+    fireEvent(element, "show-dialog", {
+      dialogTag: "confirm-event-dialog-box",
+      dialogImport: loadGenericDialog,
+      dialogParams: {
+        ...dialogParams,
+        confirm: (thisAndFuture: RecurrenceRange) => {
+          resolve(thisAndFuture);
+        },
+        cancel: () => {
+          resolve(undefined);
+        },
```

Comment:
Why hardcode this here?

Reply:
My goal is just to simplify the caller logic so this works:

```
const ....  = await showConfirmEventDialog(this, ...);
if (.....) {   // Cancel
    return
} else {
  try {
    await deleteCalendarEvent(
      ...  // delete this event, or delete this and future
    );
  } catch (err: any) {
...
```

The approach I took was copied from an existing dialog, but i don't like it. Can you give me a suggestion on another example to look at or a better structure?

Reply:
You copied it from the confirm dialog I guess?

What you miss here is the
```ts
    const origCancel = dialogParams.cancel;
    const origConfirm = dialogParams.confirm;
```

and then all those in your overwrite of `confirm` and `cancel`

Reply:
Ah yes, i removed that since it was unused, but now see why thats confusing. Fixed.